src/App.jsx

```diff
@@ -14,45 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function setOrder(goods, fild, revers) {
+  const prepereGoods = [...goods];
+
+  if (fild) {
+    prepereGoods.sort((good1, good2) => {
+      switch (fild) {
+        case SORT_BY_LENGTH:
+          return good1[fild] - good2[fild];
+
+        case SORT_BY_ALPHABET:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (revers) {
+    prepereGoods.reverse();
+  }
+
+  return prepereGoods;
+}
+
+const SORT_BY_LENGTH = 'length';
+const SORT_BY_ALPHABET = 'alphabet';
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [fildOrder, setFildOrder] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visiblGoods = setOrder(goodsFromServer, fildOrder, isReversed);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': fildOrder !== SORT_BY_ALPHABET,
+          })}
+          onClick={() => setFildOrder(SORT_BY_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': fildOrder !== SORT_BY_LENGTH,
+          })}
+          onClick={() => setFildOrder(SORT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': isReversed !== true,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
```

Comment:
it's better to move this handler click to an own function
```
const handleReverseClick = () => {
  setIsReversed(prev => !prev)
}
```
```suggestion
          onClick={handleReverseClick}
```